### PR TITLE
use grad2 in node_regr

### PR DESCRIPTION
**Pull request by willwheelera**
in _node_regr we have the line d2 += np.sum(np.abs(wf.gradient(e, configs.electron(e))) ** 2, axis=0). this is the same quantity grad2 that is computed by energy.kinetic after the recent DMC upgrade. 

This change reuses grad2 instead of computing all the gradients again.

optimize_ortho.collect_overlap_data calls pgrad._node_regr directly, as well as callign pgrad.avg. grad2 is being computed twice here, but I haven't figured out how to reuse it, so it's recomputed in collect_overlap_data for now.

### CODE REVIEW

File: pyqmc/optimize_ortho.py [hunk 1]
```diff
@@ -50,7 +50,11 @@ def collect_overlap_data(wfs, configs, pgrad):
     ) / len(ref)
 
     dppsi = pgrad.transform.serialize_gradients(wfs[-1].pgradient())
-    node_cut, f = pgrad._node_regr(configs, wfs[-1])
+    d2 = 0.0
```
Comment (lkwagner) [inline pyqmc/optimize_ortho.py:53]: can you get this from the line below? 
`    dat = pgrad.avg(configs, wfs[-1], weight[-1])`

Comment (willwheelera) [reply pyqmc/optimize_ortho.py:53]: I haven't figured out how. pgrad.avg averages over configs, but we need that axis for averaging overlap_gradient

Comment (lkwagner) [reply pyqmc/optimize_ortho.py:53]: you can just call it and not do the averaging, then. The only issue is that then the matrix becomes very large.

**Comment by lkwagner** [comment]
Did you give up on this? I don't recall a resolution to our discussion.

**Pull request merged**